perf/src/noprotection.rs

```diff
@@ -129,6 +130,10 @@ impl crypto::ClientConfig for NoProtectionClientConfig {
 
         Ok(Box::new(NoProtectionSession::new(tls)))
     }
+
+    fn as_any(&self) -> &dyn Any {
```

Comment:
`&dyn Any` + downcasting seems relatively ugly. Why can't we do a `&dyn crypto::ClientConfig` thing here?

Reply:
Thanks @djc ... unfortunately, I'm still a bit of a rust newbie and all the downcasting examples I've seen use an intermediate conversion to Any. My attempts at hacking around it have so far been unsuccessful. Do you have an example to illustrate your suggestion?

Reply:
Nevermind, I think I found a work-around that no longer requires downcasting.